src/bp-core/classes/class-bp-component.php

```diff
@@ -621,7 +622,7 @@ public function setup_actions() {
 		add_action( 'bp_generate_rewrite_rules', array( $this, 'generate_rewrite_rules' ), 10 );
 
 		// Register BP REST Endpoints.
-		if ( bp_rest_in_buddypress() && bp_rest_api_is_available() ) {
+		if ( bp_rest_api_is_available() ) {
```

Comment:
```suggestion
		if ( bp_rest_api_is_available() && ! bp_rest_is_plugin_active() ) {
```

It could be useful to avoid loading the built in BP REST API  if in the future we need to provide a way to load a previous version of the BP REST API.